### DISCUSSION TITLE
Pipe sqlite-utils in-memory db to datasette?

### DISCUSSION THREAD

**Discussion by nielsbom** [simonw/datasette#1637] [Q&A] [answered]
1. `sqlite-utils memory` creates a query-able in-memory sqlite database.
2. I think the `datasette` command can connect to an in-memory database.

Would it be possible to pipe (or otherwise connect) the sqlite-utils in-memory database into datasette?

**Comment by nielsbom** [comment]
For people wanting the same thing: I've created this Fish-shell script to get a single-command datasette-view on a .csv
```
function explore_csv
  sqlite-utils insert data.db dbname $argv[1] --csv
  datasette serve data.db -o
  rm data.db 
end
```

**Comment by simonw** [comment, marked as answer]
This is a really interesting idea! I don't *think* it's possible to do this, because the in-memory database exists inside the `sqlite-utils` process which means a separate Datasette process would be unable to see it.

I'm really interested in making it easier to explore things like CSV files using Datasette without having to first write a SQLite database file to disk though.

[Datasette Desktop](https://datasette.io/desktop) can do this right now - it has a "open CSV" option which opens the CSV directly into an in-memory database so you can start browsing it.

I also have a concept for a [datasette insert](https://github.com/simonw/datasette/issues/1160) mechanism, which would allow direct imports of formats such as CSV and would let plugins extend that to other formats as well. It's still mostly just a concept though.

**Comment by nielsbom** [comment]
Thanks for replying.

I'll keep an eye out for updates and will keep using my shell script for now, it works pretty well :-)

**Comment by plpxsk** [comment]
My data too is often a CSV, or a collection of CSVs, which I want to visualize with `datasette`.

The first step though is always to run `csvs-to-sqlite`.

I think something like `datasette file.csv` or `datasette folder_of_csvs` would be neat to have, for those of us who start with CSVs
